library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/async/RecordedDataQueueHandler.kt

```diff
@@ -0,0 +1,204 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.async
+
+import androidx.annotation.MainThread
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.WorkerThread
+import com.datadog.android.sessionreplay.internal.processor.RecordedDataProcessor
+import com.datadog.android.sessionreplay.internal.processor.RumContextDataHandler
+import com.datadog.android.sessionreplay.internal.recorder.SystemInformation
+import com.datadog.android.sessionreplay.internal.utils.TimeProvider
+import com.datadog.android.sessionreplay.model.MobileSegment
+import java.lang.ClassCastException
+import java.lang.NullPointerException
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.LinkedBlockingDeque
+import java.util.concurrent.RejectedExecutionException
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+
+/**
+ * Responsible for storing the Snapshot and Interaction events in a queue.
+ * Allows for asynchronous enrichment, which still preserving the event order.
+ * The items are added to the queue from the main thread and processed on a background thread.
+ */
+internal class RecordedDataQueueHandler {
+
+    private var executorService: ExecutorService
+    private var processor: RecordedDataProcessor
+    private var rumContextDataHandler: RumContextDataHandler
+    private var timeProvider: TimeProvider
+
+    internal constructor(
+        processor: RecordedDataProcessor,
+        rumContextDataHandler: RumContextDataHandler,
+        timeProvider: TimeProvider
+    ) {
+        this.processor = processor
+        this.rumContextDataHandler = rumContextDataHandler
+        this.timeProvider = timeProvider
+
+        // TODO: RUMM-0000 change to LoggingThreadPoolExecutor once V2 is merged
+        this.executorService = ThreadPoolExecutor(
+            CORE_DEFAULT_POOL_SIZE,
+            CORE_DEFAULT_POOL_SIZE,
+            THREAD_POOL_MAX_KEEP_ALIVE_MS,
+            TimeUnit.MILLISECONDS,
+            LinkedBlockingDeque()
+        )
+    }
```

Comment:
minor: probably for simplification this one can call another one:

```suggestion
    ): this(
    processor,
      rumContextDataHandler,
      timeProvider,
      ThreadPoolExecutor(
            CORE_DEFAULT_POOL_SIZE,
            CORE_DEFAULT_POOL_SIZE,
            THREAD_POOL_MAX_KEEP_ALIVE_MS,
            TimeUnit.MILLISECONDS,
            LinkedBlockingDeque()
        )
      )
```